tests/build_spec_generator/reproducible_central/compare_rc_build_spec.py

```diff
@@ -0,0 +1,165 @@
+# Copyright (c) 2025 - 2025, Oracle and/or its affiliates. All rights reserved.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/.
+
+"""This script compares 2 Reproducible Central Buildspec files."""
+
+import logging
+import os
+import sys
+from collections.abc import Callable
+
+CompareFn = Callable[[object, object], bool]
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+logging.basicConfig(format="[%(filename)s:%(lineno)s %(tag)s] %(message)s")
+
+
+def log_with_tag(tag: str) -> Callable[[str], None]:
+    """Generate a log function that prints the name of the file and a tag at the beginning of each line."""
+
+    def log_fn(msg: str) -> None:
+        logger.info(msg, extra={"tag": tag})
+
+    return log_fn
+
+
+log_info = log_with_tag("INFO")
+log_err = log_with_tag("ERROR")
+log_failed = log_with_tag("FAILED")
+log_passed = log_with_tag("PASSED")
+
+
+def log_diff_str(name: str, result: str, expected: str) -> None:
+    """Pretty-print the diff of two Python strings."""
+    output = [
+        f"'{name}'",
+        *("----  Result  ---", f"{result}"),
+        *("---- Expected ---", f"{expected}"),
+        "-----------------",
+    ]
+    log_info("\n".join(output))
+
+
+def skip_compare(_result: object, _expected: object) -> bool:
+    """Return ``True`` always.
+
+    This compare function is used when we want to skip comparing a field.
+    """
+    return True
+
+
+def compare_rc_build_spec(
```

Comment:
Most of the logic of this function is taken from https://github.com/oracle/macaron/blob/6a712af1ebdcb435bd5b7199dc1b4f5473663090/tests/vsa/compare_vsa.py

I think the comparing functions within `compare_vsa.py` could be refactored out into a `tests_util.py` module so that all "compare" scripts used in integration tests ([here](https://github.com/oracle/macaron/blob/b63eb9be22da84e595339aeaf291c81bd674d630/tests/integration/run.py#L77-L84)) could use if needed. 
Please let me know if this should be done in a subsequent PR or this PR.

Reply:
Either is fine for me.